File: 01-01-2025_Print_Anagrams_Together.cpp
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>
#include <unordered_map>
using namespace std;

class Solution {
public:
    vector<vector<string>> anagrams(vector<string>& arr) {
        unordered_map<string, vector<string>> map;
        
        // Grouping anagrams
        for (string str : arr) {
            string sortedStr = str;
            sort(sortedStr.begin(), sortedStr.end());
            map[sortedStr].push_back(str);
        }
        
        // Collecting results
        vector<vector<string>> result;
        for (auto& entry : map) {
            result.push_back(entry.second);
        }
        
        // Sorting groups lexicographically
        sort(result.begin(), result.end());
        return result;
    }
};

File: 01-01-2025_Print_Anagrams_Together.java
class Solution {
    public List<List<String>> anagrams(String[] arr) {
        Map<String, List<String>> map = new HashMap<>();
        
        // Grouping anagrams
        for (String str : arr) {
            char[] charArray = str.toCharArray();
            Arrays.sort(charArray);
            String sortedStr = new String(charArray);
            
            if (!map.containsKey(sortedStr)) {
                map.put(sortedStr, new ArrayList<>());
            }
            map.get(sortedStr).add(str);
        }
        
        // Collecting results
        List<List<String>> result = new ArrayList<>(map.values());
        
        // Sorting groups lexicographically
        result.sort((a, b) -> a.get(0).compareTo(b.get(0)));
        return result;
    }
}
